### PR TITLE
uint256 nonce to uint64 nonce

### PR DESCRIPTION
**Pull request by axchu**
### Fixes
- Fixes # https://github.com/ethereum-optimism/roadmap/issues/194

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/libraries/rlp/Lib_RLPReader.sol [hunk 1]
```diff
@@ -292,6 +292,42 @@ library Lib_RLPReader {
...
+        internal
+        pure
+        returns (
+            uint64
+        )
+    {
+        return uint64(uint256(readBytes32(_in)));
+    }
+
+        /**
```
Comment (ben-chain) [inline contracts/optimistic-ethereum/libraries/rlp/Lib_RLPReader.sol:312]: indent typo

File: contracts/optimistic-ethereum/libraries/rlp/Lib_RLPReader.sol [hunk 2]
```diff
@@ -292,6 +292,42 @@ library Lib_RLPReader {
...
+    function readUint64(
+        RLPItem memory _in
+    )
+        internal
+        pure
+        returns (
+            uint64
+        )
+    {
+        return uint64(uint256(readBytes32(_in)));
```
Comment (ben-chain) [inline contracts/optimistic-ethereum/libraries/rlp/Lib_RLPReader.sol:309]: Can we add test cases alongside the existing test JSON for `Lib_RLPReader` just to keep up the coverage?  Looks fine but still good practice.

**Review by ben-chain — DISMISSED** [review]
Looks good!  Approving for future merge, but did have one request for a couple of small unit tests on the new library methods.

Let's hold off on merging this one until we know we're good on the geth side.  Will figure that out.

**Comment by kfichter** [comment]
LGTM.

**Comment by axchu** [comment]
@ben-chain wants to leave this open b/c of downstream changes

**Pull request closed without merging**